core/schema/modulesource.go

```diff
@@ -514,6 +514,106 @@ func (s *moduleSchema) filterUnInstalledDeps(ctx context.Context, bk *buildkit.C
 	return effectiveDependencies, nil
 }
 
+func (s *moduleSchema) applyDepUpdates(ctx context.Context, bk *buildkit.Client, currentDeps []*modules.ModuleConfigDependency, updateList []string, updateAll bool) ([]*modules.ModuleConfigDependency, error) {
+	if len(updateList) == 0 && !updateAll {
+		return currentDeps, nil
+	}
+
+	updatesDepsMap := map[string]parsedRefString{}
+	toBeUpdated := map[string]string{}
+	for _, dep := range updateList {
+		depParsed := parseRefString(ctx, bk, dep)
+
+		// for scenario where user tries to update using name
+		var cleanDepPath = filepath.Clean(depParsed.modPath)
+
+		updatesDepsMap[cleanDepPath] = depParsed
+		toBeUpdated[cleanDepPath] = dep
```

Comment:
Instead of having two maps here, the `parsedRefString` should instead include the `original` string it was parsed from.